### PR TITLE
Employee qualification entity crud

### PR DESCRIPTION
**Pull request by dvanhout89**
- EmployeeQualification Services
- EmployeeQualification Controllers
- EmployeeQualificaiton Entities
- EmployeeQualification Models / Enums

### CODE REVIEW

**Review by ReinoRabbit — APPROVED** [review]

**Review by madwizzard — DISMISSED** [review]

File: HRIS.Services.Tests/Services/EmployeeQualificationServiceUnitTests.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+using Moq;
+using RR.UnitOfWork;
+using RR.UnitOfWork.Entities.HRIS;
+using Xunit;
+
+namespace HRIS.Services.Tests.Services
+{
+    public class EmployeeQualificationServiceUnitTest
+    {
+
```
Comment (tbddorothy) [inline HRIS.Services.Tests/Services/EmployeeQualificationServiceUnitTests.cs:13]: Why is the file here if the class is empty?

File: HRIS.Models/EmployeeQualificationDto.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+public class EmployeeQualificationDto
+{
+    public int Id { get; set; }
+    public int EmployeeId { get; set; }
+
+    [Required]
+    public HighestQualification HighestQualification { get; set; }
+    [Required]
+    public string? School { get; set; }
```
Comment (tbddorothy) [inline HRIS.Models/EmployeeQualificationDto.cs:14]: Does this ? not make the attribute nullable?

File: HRIS.Services.Tests/Services/EmployeeQualificationServiceUnitTests.cs (hunk 1 above)
Comment (dvanhout89) [reply HRIS.Services.Tests/Services/EmployeeQualificationServiceUnitTests.cs:13]: You told me yesterday to make a pr even if the unit test is not done. Should I delete the file?

File: RR.App.Tests/Controllers/HRIS/EmployeeQualificationControllerUnitTests.cs [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+        var okResult = Assert.IsType<OkObjectResult>(result);
+        var returnValue = Assert.IsType<EmployeeQualificationDto>(okResult.Value);
+        Assert.Equal(deletedQualification, returnValue);
+    }
+
+    [Fact]
+    public async Task DeleteEmployeeQualificationReturnsNotFoundWhenKeyNotFoundExceptionThrown()
+    {
+        var id = 1;
+        var exceptionMessage = "Qualification not found";
```
Comment (tbddorothy) [inline RR.App.Tests/Controllers/HRIS/EmployeeQualificationControllerUnitTests.cs:193]:
```suggestion
        var expectedExceptionMessage = "Qualification not found";
```

File: RR.App.Tests/Controllers/HRIS/EmployeeQualificationControllerUnitTests.cs [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+
+        var notFoundResult = Assert.IsType<NotFoundObjectResult>(result);
+        Assert.Equal(exceptionMessage, notFoundResult.Value);
+    }
+
+    [Fact]
+    public async Task DeleteEmployeeQualificationReturnsStatusCode500OnException()
+    {
+        var id = 1;
+        var exceptionMessage = "An unexpected error occurred";
```
Comment (tbddorothy) [inline RR.App.Tests/Controllers/HRIS/EmployeeQualificationControllerUnitTests.cs:208]:
```suggestion
        var expectedExceptionMessage = "An unexpected error occurred";
```

File: HRIS.Models/EmployeeQualificationDto.cs (hunk 2 above)
Comment (dvanhout89) [reply HRIS.Models/EmployeeQualificationDto.cs:14]: which line?

File: RR.Tests.Data/Models/HRIS/EmployeeQualificationTestData.cs [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+public static class EmployeeQualificationTestData
+{
+    //public static EmployeeQualificationDto BachelorQualification => new EmployeeQualificationDto
+    //{
+    //    Id = 1,
+    //    EmployeeId = 1,
+    //    Qualification = "Bachelor's Degree",
+    //    School = "University of Pretoria",
+    //    Degree = (DegreeType)Enum.Parse(typeof(DegreeType), "Bachelor"),
```
Comment (tbddorothy) [inline RR.Tests.Data/Models/HRIS/EmployeeQualificationTestData.cs:14]: Ghost code?

File: RR.UnitOfWork.Tests/Entities/EmployeeQualificationUnitTests.cs [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+using HRIS.Models.Enums;
+using HRIS.Models.Enums.QualificationEnums;
+using RR.UnitOfWork.Entities.HRIS;
+using Xunit;
+
+namespace RR.UnitOfWork.Tests.Entities;
+
+public class EmployeeQualificationUnitTests
+{
+
```
Comment (tbddorothy) [inline RR.UnitOfWork.Tests/Entities/EmployeeQualificationUnitTests.cs:11]: Remove empty classes please

File: HRIS.Models/EmployeeQualificationDto.cs (hunk 2 above)
Comment (dvanhout89) [reply HRIS.Models/EmployeeQualificationDto.cs:14]: All the fields for the qualification are required.

Comment (dvanhout89) [reply HRIS.Models/EmployeeQualificationDto.cs:14]: I see the error in my code. I will make an update to make all the fields non-nullable values.

File: RR.Tests.Data/Models/HRIS/EmployeeQualificationTestData.cs (hunk 5 above)
Comment (dvanhout89) [reply RR.Tests.Data/Models/HRIS/EmployeeQualificationTestData.cs:14]: Its for the Unit tests. Will remove it asap

**Review by tbddorothy — APPROVED** [review]

**Review by madwizzard — APPROVED** [review]

**Review by ReinoRabbit — APPROVED** [review]

**Pull request merged**